x-pack/functionbeat/docs/overview.asciidoc

```diff
@@ -10,12 +10,89 @@
 deploy on your serverless environment to collect events generated by cloud
 services and ship the events to {es}.
 
-This version supports deploying {beatname_uc} as an AWS Lambda service and
+Version {version} supports deploying {beatname_uc} as an AWS Lambda service and
 responds to the triggers defined for the following event sources:
 
 * https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/WhatIsCloudWatchLogs.html[CloudWatch Logs]
 * https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/welcome.html[Amazon Simple Queue Service (SQS)]
 * https://docs.aws.amazon.com/kinesis/latest/APIReference/Welcome.html[Kinesis]
 
+image::./images/diagram-functionbeat-architecture.svg["{beatname_uc} collects events generated by cloud services"]
+
 include::{libbeat-dir}/docs/shared-libbeat-description.asciidoc[]
 
+The following sections explore some common use cases for {beatname_uc}:
+
+* <<monitor-cloud-deployments>>
+* <<event-drive-processing>>
+
+[float]
+[[monitor-cloud-deployments]]
+=== Monitor cloud deployments
+
+You can deploy {beatname_uc} on your serverless environment to collect logs and
+metrics generated by cloud services and stream the data to the {stack} for
+centralized analytics.
+
+[float]
+==== Monitor AWS services with CloudWatch logs
+
+You can deploy {beatname_uc} as a Lambda function on AWS to receive events from
+a Cloudwatch Log group, extract and structure the relevant fields, then stream
+the events to {es}.
+
+// REVIEWERS: I decided to talk about the underpinning technologies
+// here (like dissect processor, filtering, and ingest node) to help users
+// understand the flow of data. I've tried to put enough descriptive elements
+// to avoid confusing new users. I think some simple diagrams that show these
+// components will be useful to help users understand the parts involved.
+
+The processing pipeline for this use case typically looks like this:
+
+. {beatname_uc} runs as a Lambda function on AWS and reads the data stream from
+a Cloudwatch Log group.
```

Comment:
@dedemorton FYI we're soon going to be enabled the use case here where Cloudwatch Logs can feed a Kinesis queue and Functionbeat will consume directly from Kinesis. This is important for organizations that may have multiple AWS subscriptions that all need to centralize their logs. We're still missing a few things to make this work such as gzip support: https://github.com/elastic/beats/pull/12119

Reply:
Nevermind, should have kept reading. :)